fw/cache.c

```diff
@@ -3145,7 +3179,10 @@ tfw_cache_start(void)
 	if (!(cache_cfg.cache || g_vhost->cache_purge))
 		return 0;
 
-	for_each_node_with_cpus(i) {
+	if (!tfw_init_node_cpus())
```

Comment:
You close db in case of SUCCESS!, more over you should not go to close_db because db is not open! (I catch three different kernel bugs here!)

Reply:
if ((r = tfw_init_node_cpus())) is good